src/marker.js

```diff
@@ -15,52 +20,57 @@ export default class Marker extends React.Component {
     anchor: OverlayPropTypes.anchor,
     offset: OverlayPropTypes.offset,
     onClick: PropTypes.func,
-  }
+    style: PropTypes.object,
+  };
 
   static defaultProps = {
     offset: 0,
-  }
+    onClick: (...args) => args,
+  };
 
-  state = {
-  }
+  state = {};
 
-  setContainer = (el) => {
-    this.container = el;
-  }
+  setContainer = el => {
+    if (el) {
+      this.container = el;
+    }
+  };
 
   handleMapMove = () => {
     this.setState(overlayState(this.props, this.context, this.container));
-  }
-
-  handleMapMoveEnd = () => {
-    this.setState(overlayState(this.props, this.context, this.container));
-  }
+  };
 
   componentDidMount() {
     const { map } = this.context;
+
     map.on('move', this.handleMapMove);
-    map.on('moveend', this.handleMapMoveEnd);
-    this.setState(overlayState(this.props, this.context, this.container));
+    map.on('moveend', this.handleMapMove);
+
+    this.handleMapMove();
   }
 
   componentWillUnmount() {
     const { map } = this.context;
+
     map.off('move', this.handleMapMove);
-    map.off('moveend', this.handleMapMoveEnd);
+    map.off('moveend', this.handleMapMove);
   }
 
   render() {
-    const onClick = this.props.onClick || (() => {});
-    const style = {
-      transform: overlayTransform(this.state),
-      zIndex: 2,
-      cursor: (this.props.onClick ? 'pointer' : 'auto'),
+    const { onClick, style } = this.props;
+
+    const finalStyle = {
```

Comment:
I think what got lost with this change is the logic to assign cursor: 'pointer' only if there was this.props.onClick passed. Looks like now style needs to be explicitly passed in, which is cool. However the default may need to be cursor: 'auto' since onClick is not required, and without optional properties the marker should not appear to be clickable?